[Bug]: Model outputs are always '!!!!!!!!!!!!!!'

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Collecting environment information...
==============================
        System Info
==============================
OS                           : Ubuntu 20.04.6 LTS (x86_64)
GCC version                  : (Ubuntu 11.4.0-2ubuntu1~20.04) 11.4.0
Clang version                : 10.0.0-4ubuntu1 
CMake version                : version 4.1.0
Libc version                 : glibc-2.31

==============================
       PyTorch Info
==============================
PyTorch version              : 2.7.1+cu128
Is debug build               : False
CUDA used to build PyTorch   : 12.8
ROCM used to build PyTorch   : N/A

==============================
      Python Environment
==============================
Python version               : 3.12.9 (main, Feb  5 2025, 19:10:45) [Clang 19.1.6 ] (64-bit runtime)
Python platform              : Linux-5.15.0-91-generic-x86_64-with-glibc2.31

==============================
       CUDA / GPU Info
==============================
Is CUDA available            : True
CUDA runtime version         : 12.1.66
CUDA_MODULE_LOADING set to   : LAZY
GPU models and configuration : 
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090
GPU 2: NVIDIA GeForce RTX 3090
GPU 3: NVIDIA GeForce RTX 3090
GPU 4: NVIDIA GeForce RTX 3090
GPU 5: NVIDIA GeForce RTX 3090
GPU 6: NVIDIA GeForce RTX 3090
GPU 7: NVIDIA GeForce RTX 3090
GPU 8: NVIDIA GeForce RTX 3090
GPU 9: NVIDIA GeForce RTX 3090

Nvidia driver version        : 530.30.02
cuDNN version                : Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.9.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv.so.9.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn.so.9.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_engines_precompiled.so.9.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_engines_runtime_compiled.so.9.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_graph.so.9.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_heuristic.so.9.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops.so.9.6.0
/usr/local/cuda-12.1/targets/x86_64-linux/lib/libcudnn.so.8.9.7
/usr/local/cuda-12.1/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.9.7
/usr/local/cuda-12.1/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.9.7
/usr/local/cuda-12.1/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.9.7
/usr/local/cuda-12.1/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.9.7
/usr/local/cuda-12.1/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.9.7
/usr/local/cuda-12.1/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.9.7
HIP runtime version          : N/A
MIOpen runtime version       : N/A
Is XNNPACK available         : True

==============================
          CPU Info
==============================
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      52 bits physical, 57 bits virtual
CPU(s):                             128
On-line CPU(s) list:                0-127
Thread(s) per core:                 2
Core(s) per socket:                 32
Socket(s):                          2
NUMA node(s):                       2
Vendor ID:                          GenuineIntel
CPU family:                         6
Model:                              106
Model name:                         Intel(R) Xeon(R) Platinum 8358 CPU @ 2.60GHz
Stepping:                           6
CPU MHz:                            800.001
CPU max MHz:                        3400.0000
CPU min MHz:                        800.0000
BogoMIPS:                           5200.00
Virtualization:                     VT-x
L1d cache:                          3 MiB
L1i cache:                          2 MiB
L2 cache:                           80 MiB
L3 cache:                           96 MiB
NUMA node0 CPU(s):                  0-31,64-95
NUMA node1 CPU(s):                  32-63,96-127
Vulnerability Gather data sampling: Mitigation; Microcode
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities

==============================
Versions of relevant libraries
==============================
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu12==12.8.3.14
[pip3] nvidia-cuda-cupti-cu12==12.8.57
[pip3] nvidia-cuda-nvrtc-cu12==12.8.61
[pip3] nvidia-cuda-runtime-cu12==12.8.57
[pip3] nvidia-cudnn-cu12==9.7.1.26
[pip3] nvidia-cudnn-frontend==1.13.0
[pip3] nvidia-cufft-cu12==11.3.3.41
[pip3] nvidia-cufile-cu12==1.13.0.11
[pip3] nvidia-curand-cu12==10.3.9.55
[pip3] nvidia-cusolver-cu12==11.7.2.55
[pip3] nvidia-cusparse-cu12==12.5.7.53
[pip3] nvidia-cusparselt-cu12==0.6.3
[pip3] nvidia-ml-py==12.575.51
[pip3] nvidia-nccl-cu12==2.26.2
[pip3] nvidia-nvjitlink-cu12==12.8.61
[pip3] nvidia-nvtx-cu12==12.8.55
[pip3] onnx==1.18.0
[pip3] onnxruntime-gpu==1.22.0
[pip3] open_clip_torch==3.1.0
[pip3] pynvml==12.0.0
[pip3] pyzmq==27.0.1
[pip3] sentence-transformers==5.1.0
[pip3] torch==2.7.1+cu128
[pip3] torchao==0.12.0+cu128
[pip3] torchaudio==2.7.1+cu128
[pip3] torchdata==0.11.0
[pip3] torchtitan==0.1.0
[pip3] torchvision==0.22.1+cu128
[pip3] transformers==4.55.2
[pip3] triton==3.3.1
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] nvidia-ml-py              12.535.133               pypi_0    pypi
[conda] transformers              4.55.1                   pypi_0    pypi

==============================
         vLLM Info
==============================
ROCM Version                 : Could not collect
Neuron SDK Version           : N/A
vLLM Version                 : 0.10.1.dev626+g1d20c3471 (git sha: 1d20c3471)
vLLM Build Flags:
  CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology:
        GPU0    GPU1    GPU2    GPU3    GPU4    GPU5    GPU6    GPU7    GPU8    GPU9    CPU Affinity    NUMA Affinity
GPU0     X      PXB     PXB     PXB     PXB     SYS     SYS     SYS     SYS     SYS     0-31,64-95      0
GPU1    PXB      X      PIX     PXB     PXB     SYS     SYS     SYS     SYS     SYS     0-31,64-95      0
GPU2    PXB     PIX      X      PXB     PXB     SYS     SYS     SYS     SYS     SYS     0-31,64-95      0
GPU3    PXB     PXB     PXB      X      PIX     SYS     SYS     SYS     SYS     SYS     0-31,64-95      0
GPU4    PXB     PXB     PXB     PIX      X      SYS     SYS     SYS     SYS     SYS     0-31,64-95      0
GPU5    SYS     SYS     SYS     SYS     SYS      X      PXB     PXB     PXB     PXB     32-63,96-127    1
GPU6    SYS     SYS     SYS     SYS     SYS     PXB      X      PIX     PXB     PXB     32-63,96-127    1
GPU7    SYS     SYS     SYS     SYS     SYS     PXB     PIX      X      PXB     PXB     32-63,96-127    1
GPU8    SYS     SYS     SYS     SYS     SYS     PXB     PXB     PXB      X      PIX     32-63,96-127    1
GPU9    SYS     SYS     SYS     SYS     SYS     PXB     PXB     PXB     PIX      X      32-63,96-127    1

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

==============================
     Environment Variables
==============================
LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu:/data/miniconda3/lib:/usr/local/cuda/lib64/:/usr/local/lib:/usr/lib/x86_64-linux-gnu:/data/miniconda3/lib:/usr/local/cuda/lib64/:/usr/local/lib:/usr/local/cuda-12.1/lib64:/usr/local/cuda-12.1/lib64
CUDA_HOME=/usr/local/cuda
CUDA_HOME=/usr/local/cuda
NCCL_CUMEM_ENABLE=0
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1
CUDA_MODULE_LOADING=LAZY
```

</details>


### 🐛 Describe the bug


```python
import os
os.environ["VLLM_USE_V1"] = "1"
os.environ["CUDA_VISIBLE_DEVICES"] = "0,1"
os.environ["VLLM_USE_FLASHINFER_SAMPLER"] = "0"
from vllm.config import CompilationConfig
from vllm import LLM
from transformers import AutoTokenizer

model = "/data/pretrained_models/Qwen3-30B-A3B-Thinking-2507-FP8"
tokenizer = AutoTokenizer.from_pretrained(model, trust_remote_code=True)
config = CompilationConfig(
    level=3,
    compile_sizes=[1, 2, 4],
    # pass_config={
    #     "enable_async_tp": True,
    # }
)
if __name__ == "__main__":
    llm = LLM(
        model=model,
        enforce_eager=True,
        tensor_parallel_size=2,
        enable_expert_parallel=True,
        compilation_config=config,
        max_model_len=1024,
        gpu_memory_utilization=0.9,
        enable_prefix_caching=True,
        enable_chunked_prefill=True,
    )
    messages = [
        {"role": "user", "content": "你叫什么名字"}
    ]
    prompt = tokenizer.apply_chat_template(
        messages,
        tokenize=False,
        add_generation_prompt=True,
    )
    outputs = llm.generate(prompt)
    print("outputs:", outputs)

```
The output is
```
[RequestOutput(request_id=0, prompt='<|im_start|>user\n你叫什么名字<|im_end|>\n<|im_start|>assistant\n<think>\n', prompt_token_ids=[151644, 872, 198, 56568, 99882, 99245, 101419, 151645, 198, 151644, 77091, 198, 151667, 198], encoder_prompt=None, encoder_prompt_token_ids=None, prompt_logprobs=None, outputs=[CompletionOutput(index=0, text='!!!!!!!!!!!!!!!!', token_ids=[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], cumulative_logprob=None, logprobs=None, finish_reason=length, stop_reason=None)], finished=True, metrics=None, lora_request=None, num_cached_tokens=0, multi_modal_placeholders={})]
```
And if I set `os.environ["VLLM_USE_FLASHINFER_SAMPLER"] = "1"`, I will encounter an illegal memory error

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

@jeejeelee Yes. And it's caused by `self.mlp`. @noooop 
```
def forward(
        self,
        positions: torch.Tensor,
        hidden_states: torch.Tensor,
        residual: Optional[torch.Tensor],
    ) -> tuple[torch.Tensor, torch.Tensor]:
        assert not torch.isnan(hidden_states).any(), "hidden_states isnan before self attn"
        # Self Attention
        if residual is None:
            residual = hidden_states
            hidden_states = self.input_layernorm(hidden_states)
            assert not torch.isnan(hidden_states).any(), "hidden_states isnan after input_layernorm, no residual"
        else:
            hidden_states, residual = self.input_layernorm(
                hidden_states, residual)
            assert not torch.isnan(hidden_states).any(), "hidden_states isnan after input_layernorm, with residual"
        hidden_states = self.self_attn(
            positions=positions,
            hidden_states=hidden_states,
        )
        assert not torch.isnan(hidden_states).any(), "hidden_states isnan after self attn"

        # Fully Connected
        hidden_states, residual = self.post_attention_layernorm(
            hidden_states, residual)
        assert not torch.isnan(hidden_states).any(), "hidden_states isnan after post_attention_layernorm"
        hidden_states = self.mlp(hidden_states)
        assert not torch.isnan(hidden_states).any(), "hidden_states isnan after mlp"
        return hidden_states, residual
``` 
```
AssertionError: hidden_states isnan after mlp
```

--- a/vllm/model_executor/layers/quantization/fp8.py
+++ b/vllm/model_executor/layers/quantization/fp8.py
@@ -125,6 +125,10 @@ class Fp8Config(QuantizationConfig):
         ignored_layers = cls.get_from_keys_or(config, ["ignored_layers"], None)
         weight_block_size = cls.get_from_keys_or(config, ["weight_block_size"],
                                                  None)
+        if not ignored_layers:
+            ignored_layers = cls.get_from_keys_or(config,
+                                                  ["modules_to_not_convert"],
+                                                  None)
         return cls(is_checkpoint_fp8_serialized=is_checkpoint_fp8_serialized,
                    activation_scheme=activation_scheme,
                    ignored_layers=ignored_layers,

